src/asm3/template.py

```diff
@@ -31,6 +31,8 @@ def get_html_template_names(dbo: Database) -> List[str]:
 
 def update_html_template(dbo: Database, username: str, name: str, head: str, body: str, foot: str, builtin: bool = False) -> None:
     """ Creates/updates an HTML publishing template """
+    row = dbo.first_row(dbo.query("SELECT * FROM templatehtml WHERE Name = ?", (name,)))
+    sql = dbo.row_to_update_sql("templatehtml", row, "NAME")
     dbo.execute("DELETE FROM templatehtml WHERE Name = ?", [name])
     htid = dbo.insert("templatehtml", {
         "Name":     name,
```

Comment:
Wouldn't it make life much easier if we just used the same ID again? We can use it to indicate whether it's new or not and if it isn't new, delete it first and handle the write to deletion.

```
htid = dbo.query_int("SELECT ID FROM templatehtml WHERE Name=?", [name])
if htid == 0: 
    htid = dbo.get_id("templatehtml")
else:
    dbo.delete(templatehtml, htid, username)

dbo.insert("templatehtml", {
    "ID":        htid,
   "Name":   name,
... etc ...
}, generateID=False) 
```

You don't need to manually call insert_deletion then as dbo.delete does it for you.